WorkloadManagementSystem/JobWrapper/JobWrapper.py

```diff
@@ -1125,35 +1127,31 @@ def finalize(self):
     requests = self.__getRequestFiles()
     outputDataRequest = self.failoverTransfer.getRequest()
 
-    # try to send the failover request
-    res = self.sendFailoverRequest()
-    if not res['OK']:  # This means that the request could not be set (this should "almost never" happen)
-      self.__report(JobStatus.FAILED, minorStatus='Failed sending requests')
-      self.__cleanUp()
-      return 1
-
-    requestFlag = res['Value'] or len(requests) > 0 or not outputDataRequest.isEmpty()
-    if requestFlag:
-      self.log.info("Job finished with pending requests")
-      self.__report(minorStatus='Pending Requests')
-    else:
-      self.log.info("Job finished with no pending requests")
+    requestFlag = len(requests) > 0 or not outputDataRequest.isEmpty()
+    self.log.info("Job finished with%s pending requests" % ('' if requestFlag else ' no'))
 
     if self.failedFlag:
       self.log.info("Job finished with errors")
-      self.__report(JobStatus.FAILED)
-      self.__cleanUp()
-      return 1
-
-    self.log.info("Job finished successfully")
-    if requestFlag:
-      self.__report(JobStatus.COMPLETED)
+      self.__report(status=JobStatus.FAILED, minorStatus=JobMinorStatus.PENDING_REQUESTS if requestFlag else '')
     else:
-      self.__report(minorStatus='Execution Complete')
-      self.__report(JobStatus.DONE)
+      self.log.info("Job finished successfully")
+      if requestFlag:
+        self.__report(status=JobStatus.COMPLETED, minorStatus=JobMinorStatus.PENDING_REQUESTS if requestFlag else '')
+      else:
+        self.__report(status=JobStatus.DONE, minorStatus=JobMinorStatus.EXEC_COMPLETE)
```

Comment:
At this point the job operations are not yet finished, so the job status can not be Done but Completed